astropy/utils/data.py

```diff
@@ -1019,90 +1019,103 @@ def download_file(remote_url, cache=False, show_progress=True, timeout=None):
     else:
         url_key = remote_url
 
-    try:
-        if cache:
-            # We don't need to acquire the lock here, since we are only reading
-            with _open_shelve(urlmapfn, True) as url2hash:
-                if url_key in url2hash:
-                    return url2hash[url_key]
+    url_key_list = url_key.split(',')
```

Comment:
Sure, the problematic part is in L193, and if we go with the list we need to move the dealings with the iterables up there, too. And I don't think the extra complexity worth it, unless we want to be able to mix and match file type objects with urls in ``get_readable_fileobj``. I'm not sure whether there is any use case for that though.

Let's see what @taldcroft says.